Sources/Hummingbird/Server/Application+HTTPResponder.swift

```diff
@@ -55,11 +56,24 @@ extension HBApplication {
         }
     }
 
-    /// Context object for Channel to be provided to HBRequest
-    struct ChannelRequestContext: HBChannelContextProtocol {
-        let channel: Channel
-        var eventLoop: EventLoop { return self.channel.eventLoop }
-        var allocator: ByteBufferAllocator { return self.channel.allocator }
-        var remoteAddress: SocketAddress? { return self.channel.remoteAddress }
+    public static func loggerWithRequestId(_ logger: Logger) -> Logger {
+        let requestId = globalRequestID.loadThenWrappingIncrement(by: 1, ordering: .relaxed)
+        return logger.with(metadataKey: "hb_id", value: .stringConvertible(requestId))
     }
 }
+
+extension Logger {
+    /// Create new Logger with additional metadata value
+    /// - Parameters:
+    ///   - metadataKey: Metadata key
+    ///   - value: Metadata value
+    /// - Returns: Logger
+    func with(metadataKey: String, value: MetadataValue) -> Logger {
```

Comment:
Is there a better name for `with`?

Reply:
I think in this case it is a valid name. The parameter name makes it clear what is happening. B is A with metadata.